# 상권변화지표

In [1]:
import pandas as pd
change17 = pd.read_csv('서울시 상권분석서비스(상권-상권변화지표) 2017.csv', encoding='UTF-8')
change18 = pd.read_csv('서울시 상권분석서비스(상권-상권변화지표) 2018.csv', encoding='UTF-8')
change19 = pd.read_csv('서울시 상권분석서비스(상권-상권변화지표) 2019.csv', encoding='UTF-8')
change20 = pd.read_csv('서울시 상권분석서비스(상권-상권변화지표) 2020.csv', encoding='UTF-8')
change21 = pd.read_csv('서울시 상권분석서비스(상권-상권변화지표) 2021.csv', encoding='UTF-8')
change22 = pd.read_csv('서울시 상권분석서비스(상권-상권변화지표) 2022.csv', encoding='UTF-8')
income_spend = pd.read_csv('서울시 상권분석서비스(상권-소득소비).csv', encoding='UTF-8')

In [2]:
# change 데이터프레임의 컬럼명 비교
change17.columns == change18.columns
change17.columns == change19.columns
change17.columns == change20.columns
change17.columns == change21.columns
change17.columns == change22.columns

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True])

In [3]:
change17.columns

Index(['기준_년_코드', '기준_분기_코드', '상권_구분_코드', '상권_구분_코드_명', '상권_코드', '상권_코드_명',
       '상권_변화_지표', '상권_변화_지표_명', '운영_영업_개월_평균', '폐업_영업_개월_평균',
       '서울_운영_영업_개월_평균', '서울_폐업_영업_개월_평균'],
      dtype='object')

In [4]:
# 상권 index 통합 기준
base_columns = ['기준_년_코드', '기준_분기_코드', '상권_구분_코드', '상권_구분_코드_명', '상권_코드', '상권_코드_명']

In [5]:
# change17, change18, change19, change20, change21, change22 데이터프레임을 행 방향으로 합침
combined_df = pd.concat([change17,change18,change19,change20,change21,change22,], ignore_index=True)
combined_df

기준_년_코드  기준_분기_코드 상권_구분_코드 상권_구분_코드_명    상권_코드         상권_코드_명  \
0         2017         4        A       골목상권  2110665          곰달래도서관   
1         2017         4        U       관광특구  1001496     강남 마이스 관광특구   
2         2017         4        U       관광특구  1001491        이태원 관광특구   
3         2017         4        R       전통시장  2130269  강남시장(동작구 강남시장)   
4         2017         4        R       전통시장  2130256            신동시장   
...        ...       ...      ...        ...      ...             ...   
40015     2022         1        A       골목상권  2110541          동교초등학교   
40016     2022         1        A       골목상권  2111063      강동구립 천호도서관   
40017     2022         1        A       골목상권  2110665          곰달래도서관   
40018     2022         1        A       골목상권  2110246         월드순복음교회   
40019     2022         1        A       골목상권  2110773         당산역 13번   

      상권_변화_지표 상권_변화_지표_명  운영_영업_개월_평균  폐업_영업_개월_평균  서울_운영_영업_개월_평균  \
0           HL       상권축소          103           47             100   
1           LH       상권확장           95           60             100   
2           HH         정체          108           58             100   
3           LH       상권확장           95           62             100   
4           HH         정체          104           69             100   
...        ...        ...          ...          ...             ...   
40015       LL       다이나믹           99           50             119   
40016       HL       상권축소          120           51             119   
40017       HL       상권축소          123           51             119   
40018       HL       상권축소          124           52             119   
40019       HH         정체          123           62             119   

       서울_폐업_영업_개월_평균  
0                  52  
1                  52  
2                  52  
3                  52  
4                  52  
...               ...  
40015              55  
40016              55  
40017              55  
40018              55  
40019              55  

[40020 rows x 12 columns]

In [6]:
# change17, change18, change19, change20, change21, change22 데이터프레임의 각 행(row) 수를 더하여 모든 데이터프레임의 행 수의 총합 계산
change17.shape[0] + change18.shape[0] +change19.shape[0] +change20.shape[0] +change21.shape[0] +change22.shape[0]

40020

In [7]:
# income_spend 데이터프레임과 combined_df 데이터프레임을 base_columns을 기준으로 내부 조인하여 money 데이터프레임을 생성
money = pd.merge(income_spend, combined_df, on= base_columns, how='inner')

In [8]:
# money 데이터프레임의 정보를 출력
money.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39082 entries, 0 to 39081
Data columns (total 24 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   기준_년_코드         39082 non-null  int64  
 1   기준_분기_코드        39082 non-null  int64  
 2   상권_구분_코드        39082 non-null  object 
 3   상권_구분_코드_명      39082 non-null  object 
 4   상권_코드           39082 non-null  int64  
 5   상권_코드_명         39082 non-null  object 
 6   월_평균_소득_금액      38868 non-null  float64
 7   소득_구간_코드        38868 non-null  float64
 8   지출_총금액          38283 non-null  float64
 9   식료품_지출_총금액      38283 non-null  float64
 10  의류_신발_지출_총금액    38283 non-null  float64
 11  생활용품_지출_총금액     38283 non-null  float64
 12  의료비_지출_총금액      38283 non-null  float64
 13  교통_지출_총금액       38283 non-null  float64
 14  여가_지출_총금액       38283 non-null  float64
 15  문화_지출_총금액       38283 non-null  float64
 16  교육_지출_총금액       38283 non-null  float64
 17  유흥_지출_총금액       38283 non-null 

# null 값 존재
- 6   월_평균_소득_금액      38868 non-null  float64
- 7   소득_구간_코드        38868 non-null  float64
- 8   지출_총금액          38283 non-null  float64
- 9   식료품_지출_총금액      38283 non-null  float64
- 10  의류_신발_지출_총금액    38283 non-null  float64
- 11  생활용품_지출_총금액     38283 non-null  float64
- 12  의료비_지출_총금액      38283 non-null  float64
- 13  교통_지출_총금액       38283 non-null  float64
- 14  여가_지출_총금액       38283 non-null  float64
- 15  문화_지출_총금액       38283 non-null  float64
- 16  교육_지출_총금액       38283 non-null  float64
- 17  유흥_지출_총금액       38283 non-null  float64

In [9]:
# money 데이터프레임에서 '상권_변화_지표'와 '상권_변화_지표_명' 컬럼을 삭제
money.drop(columns=['상권_변화_지표', '상권_변화_지표_명'], inplace=True)

In [10]:
# money 데이터프레임을 csv파일로 변환
money.to_csv('money.csv',index=False)